src/app/workflow/launch/launch.component.ts

```diff
@@ -126,20 +135,45 @@ export class LaunchWorkflowComponent extends EntryTab implements OnInit, OnChang
    * @param versionName
    */
   updateWgetTestJsonString(workflowPath: string, versionName: string, descriptorType: ToolDescriptor.TypeEnum): void {
-    let toolFiles$: Observable<Array<ToolFile>>;
-    toolFiles$ = this.gA4GHFilesQuery.getToolFiles(descriptorType, [ToolFile.FileTypeEnum.TESTFILE]);
-    toolFiles$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((toolFiles: Array<ToolFile>) => {
-      if (toolFiles && toolFiles.length > 0) {
-        this.testParameterPath = toolFiles[0].path;
-      } else {
-        this.testParameterPath = null;
+    combineLatest([
+      this.gA4GHFilesQuery.getToolFiles(descriptorType, [ToolFile.FileTypeEnum.TESTFILE]),
+      this.gA4GHFilesQuery.getToolFiles(descriptorType, [ToolFile.FileTypeEnum.PRIMARYDESCRIPTOR]),
+    ]).subscribe(
+      ([toolFiles, descriptorFiles]) => {
+        // test parameter file is optional ...
+        if (toolFiles !== undefined) {
+          if (toolFiles.length > 0) {
+            this.testParameterPath = toolFiles[0].path;
+          } else {
+            this.testParameterPath = undefined;
+          }
+          this.wgetTestJsonDescription = this.launchService.getTestJsonString(
+            ga4ghWorkflowIdPrefix + workflowPath,
+            versionName,
+            descriptorType,
+            this.testParameterPath
+          );
+        }
+        if (descriptorFiles?.length > 0) {
+          // ... but primary descriptor is mandatory
```

Comment:
This may not be true of the registered Galaxy workflows currently on Dockstore, but I've seen other (invalid) workflows where there are no files associated with a version. I wonder if to be safe you shouldn't set the properties to null or undefined if this condition isn't true.

Reply:
Added, but note there's a warning " This section is only available for valid versions. A valid version requires a valid descriptor file. " that blanks out the tab anyway